fcrepo/fcrepo
# move spring xml to java config

**Jira**: https://jira.lyrasis.org/browse/FCREPO-3558

## What this does

Removes all of the config from `fcrepo-config.xml` and moves it in to Java Config, and adds properties to toggle and configure the beans.

## How to test

Try setting various properties using `-D` arguments and properties files and see that Fedora starts with the appropriate configuration set.
